gf/test/gf_new_tail_test.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 1998-2017 ALPS Collaboration. See COPYRIGHT.TXT
+ * All rights reserved. Use is subject to license terms. See LICENSE.TXT
+ * For use in publications, see ACKNOWLEDGE.TXT
+ */
+
+#include <gtest/gtest.h>
+
+#include <alps/gf_new/gf_tail.h>
+#include <alps/gf/mesh.hpp>
+
+
+namespace g=alps::gf;
+
+class GreensFunctionTailTest : public ::testing::Test
+{
+public:
+  const double beta;
+  const int nsites;
+  const int nfreq ;
+  const int nspins;
+  typedef g::greenf<std::complex<double> , g::matsubara_positive_mesh, g::index_mesh> omega_sigma_gf;
+  typedef g::greenf<std::complex<double>, alps::gf::matsubara_positive_mesh, alps::gf::index_mesh> head;
+  typedef g::greenf<double, alps::gf::index_mesh> tail;
+  omega_sigma_gf gf;
+  omega_sigma_gf gf2;
+  typedef alps::gf::matsubara_mesh<alps::gf::mesh::POSITIVE_ONLY> matsubara_mesh;
+  typedef omega_sigma_gf gf_type;
+
+  GreensFunctionTailTest():beta(10), nsites(4), nfreq(10), nspins(2),
+                   gf(matsubara_mesh(beta,nfreq),
+                      alps::gf::index_mesh(nspins)),
+                   gf2(gf) {}
+};
+
+
+TEST_F(GreensFunctionTailTest, InitializationTest){
+  typedef g::greenf<double, alps::gf::matsubara_positive_mesh, alps::gf::index_mesh, alps::gf::itime_mesh, alps::gf::legendre_mesh> head;
+  typedef g::greenf<double, alps::gf::index_mesh, alps::gf::itime_mesh, alps::gf::legendre_mesh> tail;
+  g::gf_tail<head, tail> g;
+}
+
+TEST_F(GreensFunctionTailTest, AssignTest){
+  alps::gf::matsubara_positive_mesh x(100, 10);
+  alps::gf::index_mesh y(10);
+  alps::gf::itime_mesh z(100, 10);
+  typedef g::greenf<double, alps::gf::matsubara_positive_mesh, alps::gf::index_mesh, alps::gf::itime_mesh> head;
+  typedef g::greenf<double, alps::gf::index_mesh, alps::gf::itime_mesh> tail;
+  g::greenf<double, alps::gf::matsubara_positive_mesh, alps::gf::index_mesh, alps::gf::itime_mesh> g(x,y,z);
```

Comment:
It is confusing to have the same name `g` for two different things (variable and name space).